onnxruntime/core/providers/xnnpack/math/gemm.cc

```diff
@@ -208,5 +249,23 @@ ONNX_OPERATOR_KERNEL_EX(Gemm, kOnnxDomain, 13, kXnnpackExecutionProvider,
                         KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<float>()),
                         Gemm);
 
+#ifdef XNNPACK_FP16_SUPPORTED
+ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(Gemm, kOnnxDomain, 7, 8, MLFloat16, kXnnpackExecutionProvider,
+                                        KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<MLFloat16>()),
```

Comment:
I think it's okay for the registration lines to be longer than 120 chars as it's basic plumbing code and it's more readable to not use extra lines for them.

Reply:
it's removed already